test/Target/ExportSystemC/errors.mlir

```diff
@@ -9,3 +9,48 @@ hw.module @notSupported () -> () { }
 // CHECK: <<UNSUPPORTED TYPE (!hw.inout<i2>)>>
 // expected-error @+1 {{no emission pattern found for type '!hw.inout<i2>'}}
 systemc.module @invalidType (%port0: !systemc.in<!hw.inout<i2>>) {}
+
+// -----
+
+// expected-error @+1 {{no emission pattern found for type '() -> (i32, i32)'}}
+func.func private @functionTypeMultipleResults (%a: () -> (i32, i32)) attributes {argNames=["a"]}
```

Comment:
This type of error isn't awesome. The pass writer is going to wonder why this failed. Ditto for other things in this test. (Noting, that I love the exhaustiveness of this test! I'm just wondering if there is a better way of checking this and giving feedback to the pass writer.)

Reply:
I agree that it's hard to find out what the cause of this error is. It's the same issue that the dialect conversion framework has. When you implement a set of patterns there and you made some mistake or your pattern does not match on a case you thought could not exist, you get an error stating that the legalization of a certain operation failed without stating what patterns it matched against and why each of them failed. Compared to the dialect conversion framework we don't  have a `--debug-only=dialect-conversion` kind of debug log which is very helpful in such cases. Of course, it would be possible to also implement something like this for this printer. It would also be possible to collect messages on the way on why each pattern did not match and then listing them in the final error message, but we would need to make sure that only useful messages are attached to it (not something like "did not match because the pattern only accepts `func.func` ops and it got a `func.call` op"). So the reason why this error message it not very nice is not that it is impossible, but that a proper implementation requires more than a few lines of code (requires a separate PR) and that I didn't prioritize it yet.